docs/docs/reference/contextual/extension-methods.md

```diff
@@ -32,37 +34,68 @@ assert(circle.circumference == circumference(circle))
 
 ### Translation of Calls to Extension Methods
 
-When is an extension method applicable? There are two possibilities.
+When is an extension method applicable? There are two possibilities:
 
- - An extension method is applicable if it is visible under a simple name, by being defined
+ 1. An extension method is applicable if it is visible, by being defined
```

Comment:
The "under a simple name" is actually important. That's the proper definition, the rest of the phrase are categories that satisfy the criterion

Reply:
Ah... I think now I get it, `simple name` is referring to simple name as defined by the JLS, right? Which means:
```
def test(c: c.Circle) = c.circumference
```
would not work

Reply:
either it is referring to another definition or it is a simplification (or I don't get the `visible` part). I see a lot of corner cases where qualified names work as well:
```scala
package foo {
  case class Circle(x: Double, y: Double, radius: Double)
  def (c: Circle).circumference: Double = c.radius * math.Pi * 2

  // works, even though not visible under a simple name but a qualified name
  // I guess the qualified name is turned into a simple name beforehand
  def test1(c: foo.Circle) = c.circumference

  def (i: Int) test (x: Int) = 2

  // works, even though not visible under a simple name but a qualified name
  def test2(i: scala.Int) = i test 3
}

package bar {
  def (c: foo.Circle).bar = 1.2

  // similar to test1 and still succeeds, special because top level definition in same file?
  def test3 (c: foo.Circle) = c.bar
}

package test4a {
  import foo._

  // similar to test1 and succeeds as well, probably qualified name was rewritten to simple name
  def test4a(c: foo.Circle) = c.circumference
}

package test4b {
  import foo.circumference

  // similar to test4a but this time it fails as expected.
  def test4b(c: foo.Circle) = c.cirumference
}

package baz {

  trait Baz {
    def (c: foo.Circle).baz = 2

    // similar to test1 and succeeds as well, why?
    def test6(c: foo.Circle) = c.baz
  }

  trait Qux extends baz.Baz {
    // similar to test1 and succeeds as well, does the rule of `under a simple name` not apply to inheritance?
    def test7(c: foo.Circle) = c.baz
  }
}

package quux {

  trait Quux extends baz.Baz {
    // similar to test8 but this time it fails, kind of as expected but I am confused
    // because I now thought the rule does not apply to inheritance.
    def test8(c: foo.Circle) = c.baz
  }
}

// side note
package test9 {
  trait Test9 {
    def (c: foo.Circle).test9 = 2

    // similar to test6 but this time we use the same name as the extension function and this fails
    // probably not related to simple names but I guess a bug nonetheless
    def test9(a: foo.Circle) = a.test9
  }
}
```

Reply:
// works, even though not visible under a simple name but a qualified name
    // I guess the qualified name is turned into a simple name beforehand
    def test1(c: foo.Circle) = c.circumference

No, in all these cases the names you are after are available as simple names. When I write
```
  def (c: Circle).circumference: Double = c.radius * math.Pi * 2
```
this creates a method circumference, which I can also call like this:
```
circumference(c)
```

Reply:
Ah... I confused myself: test4b does not compile due to a typo on my part and test8 does not fail but actually succeeds (still wonder why it failed at some point). Sorry for that. 

I think I finally understand what you mean with `visible under`/`available`. It is not the receiver type of the extension method (Circle) which needs to be available under a simple name but the extension method as such. Makes sense and that explains why inheritance works. 
Thanks for the feedback